sdk/log/batch.go

```diff
@@ -156,13 +156,17 @@ func (b *BatchProcessor) poll(interval time.Duration) (done chan struct{}) {
 				global.Warn("dropped log records", "dropped", d)
 			}
 
-			qLen := b.q.TryDequeue(buf, func(r []Record) bool {
-				ok := b.exporter.EnqueueExport(r)
-				if ok {
-					buf = slices.Clone(buf)
-				}
-				return ok
-			})
+			qLen := b.q.Len()
+			// don't copy data from queue unless exporter can accept more, it is very expensive
```

Comment:
This should be an opt-in `BatchProcessorOption` e.g. `WithDropping` with documentation that this changes the batch processor to drop emitted log records if the queue is full. And document that by default the oldest log records are dropped from the queue.